rfcs/0075-declarative-wrappers.md

```diff
@@ -0,0 +1,353 @@
+---
+feature: Declarative Wrappers
+start-date: (fill me in with today's date, YYYY-MM-DD)
+author: Doron Behar <me@doronbehar.com>
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: POC Implementation at [#85103](https://github.com/NixOS/nixpkgs/pull/85103).
+---
+
+# Summary
+[summary]: #summary
+
+Manage the environment of wrappers declaratively and deprecate shell based
+methods for calculating runtime environment of packages. Make wrappers a
+separate derivation so that mere changes to the environment will not trigger a
+rebuild. Make it easier to debug why env vars are added to an executable, by
+using Nix as the language to evaluate what env vars are needed, instead of not
+documented good enough and not easily debug-able shell hooks.
+
+# Motivation
+[motivation]: #motivation
+
+We have numerous issues regarding wrappers and our wrapper shell hooks. Here's
+a list of them, sorted to categories.
```

Comment:
I'd suggest using bullet points here to describe what issues this RFC resolves, before going into the details.

Reply:
> I'd suggest using bullet points here to describe what issues this RFC resolves, before going into the details.

Done.